### PR TITLE
[Ref] Remove unaligned constructors

### LINKED ISSUE #1720: Test infallible alignment errors in Fuchsia, then remove `Ref` `unaligned` constructors
With #1718, we can likely remove the `unaligned` constructors from `Ref`. However, we should first confirm that this is doable by vendoring #1718 into Fuchsia and attempting to remove all `unaligned` uses from tree.

### PR DESCRIPTION
**Pull request by joshlf**
Closes #1720

### CODE REVIEW

**Comment by joshlf** [comment]
@jswrenn Feel free to review, but I'm going to hold off on merging until the Fuchsia vendor CL [passes all of its CQ tests](https://fuchsia-review.googlesource.com/c/fuchsia/+/1122913?tab=checks).

**Comment by joshlf** [comment]
> @jswrenn Feel free to review, but I'm going to hold off on merging until the Fuchsia vendor CL [passes all of its CQ tests](https://fuchsia-review.googlesource.com/c/fuchsia/+/1122913?tab=checks).

Okay everything is passing other than asan tests, which are always flaky. I'm comfortable merging this. I'll mark as "merge when ready".

File: src/ref.rs [hunk 1]
```diff
@@ -570,278 +570,6 @@ where
...
-    /// struct ZSTy {
-    ///     leading_sized: u16,
-    ///     trailing_dst: [()],
-    /// }
-    ///
-    /// let f = Ref::<&[u8], ZSTy>::unaligned_from_bytes(&b"UU"[..]); // ⚠ Compile Error!
-    /// ```
-    #[must_use = "has no side effects"]
-    #[inline(always)]
-    pub fn unaligned_from_bytes(source: B) -> Result<Ref<B, T>, SizeError<B, T>> {
```
Comment (jswrenn) [inline src/ref.rs:610]: Should these be removals, or deprecations?

Comment (joshlf) [reply src/ref.rs:610]: Removals; they're [new as of 0.8](https://docs.rs/zerocopy/0.7.35/zerocopy/struct.Ref.html). We've already deprecated all of the 0.7 methods, which have a different naming scheme.

**Review by jswrenn — APPROVED** [review]

**Pull request merged**